### ISSUE TITLE
filter out k8s system containers.

### ISSUE THREAD

**Issue by odedpriva** [gliderlabs/logspout#404]
Hi, 

We are using logspout to collect logs from containers running in k8s cluster. 

Is there a way we could exclude k8s system logs ? or use k8s annotation mechanism ?

**Comment by develar** [comment]
It is simple.

Edit your k8s secret value to:
```
syslog://logsN.papertrailapp.com:N?filter.name=k8s_YOUR_CONTAINER_NAME_*
```

It is very important to understand that if you want to send logs from several pods, you must not specify several `filter.name` and in any case it is not supported. The same applies to `filter.labels` — default logical operation is **AND**, it means that container labels must match all specified filters.

There is easy workaround — specify several destinations:


```
syslog://logsN.papertrailapp.com:N?filter.name=k8s_YOUR_CONTAINER_NAME1_*,syslog://logsN.papertrailapp.com:N?filter.name=k8s_YOUR_CONTAINER_NAME2_*
```

It is important to use `k8s_` prefix because for each container maybe created another container (for CNI) `k8s_POD_`.

In your case container name pattern maybe different, to understand what to use you can login to your server and `docker ps --format "{{.Names}}"`.

**Comment by odedpriva** [comment]
Thanks for the answer. 

1. I'm looking for a solution that would allow me to filter **out** certain pods. ( dns , networking, etc. ) and **not** specifying pods to collect logs from. 
2. I'm using "logstash+tcp://logstash:5000", I tried to use the syntax you mentioned but it didn't seem to work. do you know if logstash plugin support this syntax?

**Comment by odedpriva** [comment]
o.k, thanks a lot for the help.

**Comment by gbolo** [comment]
Hi @odedpriva 
We recently merged #416 which adds functionality to `EXCLUDE_LABEL` to include a custom label value. This should allow you to exclude a whole kube namespace. So you have many options available here. Closing this issue.
